### PR TITLE
Api600/500 support to OS Build Plan

### PR DESCRIPTION
**Pull request by soodpr**
### Description
Api600/500 support to OS Build Plan

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [X] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [X] New endpoints supported are updated in the endpoints-support.md file.
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/image-streamer/build_plan.rb [hunk 1]
```diff
@@ -21,12 +28,27 @@
...
+}
+plan_script_class = OneviewSDK::ImageStreamer.resource_named('PlanScript', @client.api_version)
+plan_script = plan_script_class.find_by(@client, name: @plan_script1_name).first
+puts "\n#Creating a plan script with name #{options_plan_script[:name]}."
+plan_script2 = plan_script_class.new(@client, options_plan_script)
+plan_script2.create!
+plan_script2.retrieve!
+puts "\n#Plan script with name #{plan_script2['name']} and uri #{plan_script2['uri']} created successfully."
+
+# plan_script2 = plan_script_class.find_by(@client, name: @plan_script2_name).first
```
Comment (sijeesh) [inline examples/image-streamer/build_plan.rb]: required?

File: examples/image-streamer/build_plan.rb [hunk 2]
```diff
@@ -104,6 +126,9 @@
 puts "\n#Removing a build plan with id #{item2['uri']} and name #{item2['name']}:"
 item2.delete
 puts "\n#Build plan with id #{item2['uri']} and name #{item2['name']} removed successfully."
+puts "\n#Removing a build plan with id #{plan_script2['uri']} and name #{plan_script2['name']}:"
+plan_script2.delete
+puts "\n#Plan script with id #{plan_script2['uri']} and name #{plan_script2['name']} removed successfully."
 puts "\n#Removing a build plan with id #{item3['uri']} and name #{item3['name']}:"
```
Comment (madhav-bharadwaj) [inline examples/image-streamer/build_plan.rb]: Is `plan_script2` a build plan or a plan script? See lines 129 and 131.

Comment (soodpr) [reply examples/image-streamer/build_plan.rb]: its a plan script which we create in the beginning of this example

File: examples/image-streamer/build_plan.rb [hunk 3]
```diff
@@ -9,7 +9,14 @@
...
+require_relative '../_client_i3s' # Gives access to @client
+# Supported APIs:
+# - 300, 500, 600
+
+# Resources that can be created according to parameters:
+# api_version = 300 & variant = Synergy to OneviewSDK::ImageStreamer::API300::BuildPlan
+# api_version = 500 & variant = Synergy to OneviewSDK::ImageStreamer::API500::BuildPlan
+# api_version = 600 & variant = Synergy to OneviewSDK::ImageStreamer::API600::BuildPlan
 
 # Example: Create a build plan for an API300 Image Streamer
```
Comment (madhav-bharadwaj) [inline examples/image-streamer/build_plan.rb]: Can you check this comment?

File: examples/image-streamer/build_plan.rb [hunk 4]
```diff
@@ -21,12 +28,27 @@
...
-plan_script2 = OneviewSDK::ImageStreamer::API300::PlanScript.find_by(@client, name: @plan_script2_name).first
+options_plan_script = {
+  description: 'Description of this plan script',
+  name: @plan_script2_name,
+  hpProvided: false,
+  planType: 'deploy',
+  content: 'esxcli system hostname set --domain "@DomainName@"'
+}
+plan_script_class = OneviewSDK::ImageStreamer.resource_named('PlanScript', @client.api_version)
+plan_script = plan_script_class.find_by(@client, name: @plan_script1_name).first
```
Comment (madhav-bharadwaj) [inline examples/image-streamer/build_plan.rb]: Is this variable `plan_script` used anywhere?

File: examples/image-streamer/build_plan.rb (hunk 2 above)
Comment (madhav-bharadwaj) [reply examples/image-streamer/build_plan.rb]: Can you then change the puts on line 129.
`puts "\n#Removing a **build plan** with id #{plan_script2['uri']}`

File: examples/image-streamer/build_plan.rb (hunk 4 above)
Comment (soodpr) [reply examples/image-streamer/build_plan.rb]: at line 57

File: examples/image-streamer/build_plan.rb [hunk 5]
```diff
@@ -9,7 +9,14 @@
...
+# Supported APIs:
+# - 300, 500, 600
+
+# Resources that can be created according to parameters:
+# api_version = 300 & variant = Synergy to OneviewSDK::ImageStreamer::API300::BuildPlan
+# api_version = 500 & variant = Synergy to OneviewSDK::ImageStreamer::API500::BuildPlan
+# api_version = 600 & variant = Synergy to OneviewSDK::ImageStreamer::API600::BuildPlan
 
 # Example: Create a build plan for an API300 Image Streamer
 # NOTE: This will create three build plans with the following names 'Build_Plan_1', 'Build_Plan_2' and 'Build_Plan_3', then delete them.
```
Comment (madhav-bharadwaj) [inline examples/image-streamer/build_plan.rb]: This comment needs to be changes as you are not using the build plan names mentioned above

File: examples/image-streamer/build_plan.rb (hunk 4 above)
Comment (madhav-bharadwaj) [reply examples/image-streamer/build_plan.rb]: Ok.

File: examples/image-streamer/build_plan.rb (hunk 5 above)
Comment (madhav-bharadwaj) [reply examples/image-streamer/build_plan.rb]: Can ignore this,

**Pull request closed without merging**